app/models/event.rb

```diff
@@ -13,8 +13,10 @@ class Event < ActiveRecord::Base
   has_many :shipments, inverse_of: :event, dependent: :restrict_with_exception
   # Mails for an event
   has_many :event_emails
+  # Travel Sponsorships for an event
+  has_many :travel_sponsorships
   # Users for an event
-  has_many :users, through: :requests
+  has_many :users, -> { distinct }, through: :travel_sponsorships
```

Comment:
sorry but can you please tell how to define a scope for the requests in the has_many : users statement